raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2368,6 +2368,10 @@ public void close() {
         if (kafkaRaftMetrics != null) {
             kafkaRaftMetrics.close();
         }
+        if (memoryPool instanceof BatchMemoryPool) {
```

Comment:
Perhaps we can let `MemoryPool` implement `Closeable`?

By the way, was this a leak? I think the only reference to the pool is in `KafkaRaftClient`, so does that mean we were leaking `KafkaRaftClient` references?

Reply:
I have been playing whack-a-mole with cases where we leak `KafkaRaftClient` instances in tests. Most of them seem to be related to the metrics closures dragging in giant objects, and the metrics not getting deleted properly. These are existing issues not caused by this PR.

This is a bit ugly but it stabilizes the build greatly so I think we should leave it in. I didn't want to make it a close() function since I didn't want to start thinking about dealing with a closed state in all of the memory pools.

Can we leave this for now and open a JIRA?